pkg/controller/provider/controller.go

```diff
@@ -234,6 +236,30 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 		}
 	}
 
+	if provider.DeletionTimestamp != nil && k8sutil.ContainsFinalizer(provider, api.OvaProviderFinalizer) {
```

Comment:
I remembered another issue I had when looking into it, once the provider deleted the objects was no longer accessible, were you able to query its fields at this point?

Reply:
yeah, not that with a finalizer, an entity doesn't get removed until the finalizer is dropped - so when we get here, the provider still exists and has the DeletionTimestamp field set, so we can still get its properties